modules/videoio/test/test_gstreamer.cpp

```diff
@@ -73,22 +73,61 @@ Param test_data[] = {
 
 INSTANTIATE_TEST_CASE_P(videoio, videoio_gstreamer, testing::ValuesIn(test_data));
 
-TEST(Videoio_GStreamer, unsupported_pipeline)
+TEST(videoio_gstreamer, unsupported_pipeline)
 {
-    VideoCaptureAPIs apiPref = CAP_GSTREAMER;
-    if (!isBackendAvailable(apiPref, cv::videoio_registry::getStreamBackends()))
-        throw SkipTestException(cv::String("Backend is not available/disabled: ") + cv::videoio_registry::getBackendName(apiPref));
+    if (!videoio_registry::hasBackend(CAP_GSTREAMER))
+        throw SkipTestException("GStreamer backend was not found");
 
     // could not link videoconvert0 to matroskamux0, matroskamux0 can't handle caps video/x-raw, format=(string)RGBA
     std::string pipeline = "appsrc ! videoconvert ! video/x-raw, format=(string)RGBA ! matroskamux ! filesink location=test.mkv";
     Size frame_size(640, 480);
 
     VideoWriter writer;
-    EXPECT_NO_THROW(writer.open(pipeline, apiPref, 0/*fourcc*/, 30/*fps*/, frame_size, true));
+    EXPECT_NO_THROW(writer.open(pipeline, CAP_GSTREAMER, 0/*fourcc*/, 30/*fps*/, frame_size, true));
     EXPECT_FALSE(writer.isOpened());
     // no frames
     EXPECT_NO_THROW(writer.release());
 
 }
 
+TEST(videoio_gstreamer, gray16_writing)
+{
+    if (!videoio_registry::hasBackend(CAP_GSTREAMER))
+        throw SkipTestException("GStreamer backend was not found");
+
+    // generate a noise frame
+    Size frame_size(320, 240);
+    Mat frame = Mat::zeros(frame_size, CV_16U);
```

Comment:
> `Mat::zeros()` call is ambiguous as you fill the mat with random values.

Not correct.

> **zeros**

fill with zeros values.

Reply:
Fixed.

Reply:
Ambiguous isn't the right word to use, but since all the values get overwritten immediately, it is unnecessary to initialize to zero, even if it is a very fast operation. I've changed this to simply `Mat(frame_size, CV_16U)`

Reply:
Sorry for my English, ambiguous is not the right word. I mean redundant.